projects/eslint-config/plugins/liferay/lib/rules/no-aui.js

```diff
@@ -0,0 +1,122 @@
+/**
+ * SPDX-FileCopyrightText: © 2017 Liferay, Inc. <https://liferay.com>
+ * SPDX-License-Identifier: MIT
+ */
+
+const message = 'avoid using AUI, use vanilla Javascript instead';
+const MESSAGE_ARRAY = 'use the native Array class';
+const MESSAGE_EACH = 'use Array methods to iterate over items';
+const MESSAGE_GETBODY =
+	"this is a shortcut to A.one('body'), use document.querySelector";
+const MESSAGE_IO = 'use Liferay.Util.fetch';
+const MESSAGE_OBJECT = 'use the native Object class';
+const MESSAGE_MERGE = 'use Object.assign';
+const MESSAGE_MODAL = 'use Liferay.Util.openModal';
+const MESSAGE_NODE =
+	'use native methods, mostly found inside the Document interface';
+const MESSAGE_SELECTOR = 'use document.querySelector instead';
+
+module.exports = {
+	create(context) {
+		return {
+			MemberExpression(node) {
+				const nodeObjectName = node.object.name;
+				const nodePropertyName = node.property.name;
+
+				if (
```

Comment:
Hey Kreso, What do you think in using an conditional just checking `nodeObjectName === 'A'` and inside it, create a switch block checking `nodePropertyName` values?

Reply:
I had that initially @diegonvs, but that required me to have a `default` case, which is bad because we don't want to check against all usages of `A`, just these specific ones.

If I removed the `default` case, then lint would complain about not having an error message for all the cases.

So I decided to go with this ugly implementation, because the `switch case` essentially required the `default` case.